app/assets/javascripts/live_search.js

```diff
@@ -359,7 +361,15 @@
 
   LiveSearch.prototype.updateSortOptions = function updateSortOptions (results, action) {
     if (action !== this.serializeState(this.state)) { return }
-    this.updateElement(this.$sortBlock, results.sort_options_markup)
+    var currentSortOptions = this.$sortBlock.querySelector('select')
+    var newSortOptions
+
+    if (results.sort_options_markup) {
+      var EXTRACT_OPTION_TAGS = /(?<=<select .+>)(.*)(?=<\/select>)/
```

Comment:
I don't think positive lookaheads work in IE11, if you're using regex to keep compatibility with this browser. https://caniuse.com/?search=exp - the JS for the page seems to break in IE11 on Browserstack.

Instead, I think you could remove `<select>` tags from the new markup string, and then replace the `innerHTML` of the existing `<select>` with the new markup string?

e.g. https://regex101.com/r/hQ5KdJ/1

```JS
var regexThatGrabsSelects = /<\s*\/?\s*select\b[^>]*>/g
newMarkup.replace(regex, "")
existingSelect.innerHTML = newMarkup
```

Also, I think there should be a test written that ensures the regex is working

Reply:
Excellent points/suggestions 👍 I've refactored the code a bit and put the regex stuff in a function so it's a bit easier to test. Thanks @AshGDS